doc/src/EntitySpecMeta.dox

```diff
@@ -15,177 +15,309 @@
- * native types. For example, deriving from the DocumentSpecification to
- * represent a DCC tool's native file format. They should however,
- * always use the most specific well-known specification as a base. This
- * ensures maximum compatibility with other tools. Any new tool,
- * manager or vendor specific schemas should be listed in the
- * appropriate technical documentation supplied with the software.
- *
- * @warning The library of well-known OpenAssetIO base schemas is yet to
- * be ported from `FnAssetAPI`, and will be available soon.
- *
- * @section Attributes
- *
- * Entities are generally more than just a file on disk, they have
- * additional data that defines their properties or interpretation.
- *
- * In its simplest form, OpenAssetIO's @ref attributes system can be used
- * by a @ref host to store and retrieve additional data about an entity
- * from an @ref asset_management_system without the need for side-car
- * files.
- *
- * An example of this is within a tool that displays image sequences.
- * The @ref primary_string is resolved to provide the path to the images
- * on disk, and the attributes provided by the manager used to control
- * the colorspace and frame range used for playback. This can avoid
- * fragile mechanisms such as extension-based mappings and directory
- * searches.
- *
- * Extending this further, an entity's primary string does not have to
- * be a file path, or even a URL.
- *
- * If it's possible to break down the description of an entity into a
- * string, and/or set of named values, OpenAssetIO can be used directly
- * to publish or query entities of this type.
- *
- * A common application of this is its use in a timeline based
- * application to query and publish shot timing information directly to
- * a production tracking system.
- *
- * The API contract states that a @ref manager **must** guarantee to
- * store and recall any value registered by this mechanism for supported
- * specifications. This allows a host to manage arbitrary data through
- * the asset management system without any special handling by the
- * manager itself.
- *
- * If a Manager holds entities created by means other than OpenAssetIO
- * (its own interface, other APIs, etc...) then it is important that it
- * 'bridges' any attributes from the well-known OpenAssetIO names to its
- * corresponding internal name, for both set and get.
- *
- * @warning In OpenAssetIO, attributes are limited to simple constants
- * using basic types such as string, float, int and bool. Time-varying
- * data is beyond the scope of this API and should be managed through
- * alternate means such as <a href="http://opentimeline.io" target="_blank">OpenTimelineIO</a>
- * or similar.
+ * Hosts and managers are also free to make their own specifications and
+ * traits for native types. For example, createing a new specifcation by
+ * adding to the traits of `DocumentSpecification`, to represent a DCC
```

Comment:
[question] Just to double-check we're on the same page: "adding to the traits", as in something like
```python
class MyDCCSpecification(BaseSpecification):
   kTraits = DocumentSpecification.kTraits + ["MyDCC"]
```
where logically there is no relationship between `MyDCCSpecification` and `DocumentSpecification`, they just happen to have some traits in common, and in this case one happens to be a subset of the other?

Reply:
As traits are additive, there is the implied relationship that `MyDCCSpecification` is a specialisation of `DocumentSpecification` as it has all of its traits _and more_.